projects/core/src/occ/adapters/cart/converters/occ-cart-normalizer.ts

```diff
@@ -17,10 +17,17 @@ export class OccCartNormalizer implements Converter<Occ.Cart, Cart> {
     }
 
     if (source && source.entries) {
-      target.entries = source.entries.map(entry => ({
-        ...entry,
-        product: this.converter.convert(entry.product, PRODUCT_NORMALIZER),
-      }));
+      target.entries = source.entries.map(entry => {
+        const product = this.converter.convert(
+          entry.product,
+          PRODUCT_NORMALIZER
+        );
+        return {
+          ...entry,
+          key: `${product.code}_${entry.basePrice.formattedValue}_${product.name}`,
```

Comment:
if we have a configurable product, that we order twice with different engraving, this key likely to be the same. 
What about just using the `entryNumber`? Or at least put it into the mix?

Reply:
is this considered also for B2B Process ? 
In B2B process: anonymous cart entries don't have price. 
I think we can use entryNumber but only if :
1. there will be no any feature that deletes multi cart entries with single action (bulk action).
2. on two devices same cart is not changed
-
Still it won't be possible to have duplicate entries in cart like this.

Reply:
The optimal key would consist only from fields that don't change the value during cart existence. Then points mentioned by @mrSingh007 would not apply to us.

Using `entryNumber` in the mix would save us from some of the bulk action errors
(scenario cart: productA, productB, productAWithEngraving -> action: remove productA and productB -> result: productA only removed as productB key would not match).

It would however fail with some cases
(scenario cart: productA, productAWithEngraving, productAWithDifferentEngraving -> action: remove productA and productAWithEngraving -> result productA and productAWithDifferentEngraving removed).

Can we access every information about entry that makes them unique in cart endpoint? Such as this engraving? If so then everyone could override our key with something that is truly unique (eg. adding all non-default attribute values to the key). I think that's the only way to make in work with bulk actions/cross device carts.

Reply:
If OCC doesn't return any unique property the entry, then we need some heuristics to build it. We could calculate the hash of the object on the frontend side. But this can be costly, as we reload cart very often in app lifetime. Moreover, it's sensitive to entries with exactly the same properties (I'm not sure it's possible, but needs verifying).

If we can't build ideal unique key, we should expose an easy extenstion point for customers to build it themselves.
@marlass Can we move the logic of building `key` to a separate protected method?

@tobi-or-not-tobi Can we consider adding the property `guid` to cart entries in the cloud cart?